### ISSUE TITLE
Packages published with peer dependencies still containing workspace protocol (pnpm + lerna)

### ISSUE THREAD

**Issue by plouc** [lerna/lerna#3671]
## Current Behavior

I maintain [nivo](https://github.com/plouc/nivo), and I recently moved to `pnpm` and upgraded lerna to `^6.6.1`.

I published a new version of the packages a day ago, but someone reported the following error:

```
npm ERR! code EUNSUPPORTEDPROTOCOL
npm ERR! Unsupported URL Type "workspace:": workspace:*
```

I use the `workspace` protocol for all packages defined within the repo.
I checked the content of one of the published package `package.json`, and found out that while `dependencies` (and `devDependencies`) versions are properly replaced:

```
"dependencies": {
    "@nivo/scales": "0.82.0"
},
```

`peerDependencies` aren't:

```
"peerDependencies": {
    "@nivo/core": "workspace:*"
},
```

I tried to run `pnpm pack` on one of the package to see how it was handling this, and it properly replaced the `peerDependencies` version: `"@nivo/core": "0.82.0"` while `lerna` didn't.

## Expected Behavior

I would expect the `peerDependencies` to be replaced as well.

## Steps to Reproduce

Please see this [repository](https://github.com/plouc/nivo).

The command I run is `pnpm lerna publish --exact`, defined in the project's make file under the `pkgs-publish` target. 

### Configuration

<details><summary>lerna.json</summary><p>

```json
{
    "packages": ["packages/*", "api"],
    "version": "0.82.0",
    "npmClient": "pnpm",
    "useWorkspaces": true
}
```

</p></details>

### Environment

```
 Environment info:

  System:
    OS: macOS 13.2
    CPU: (10) arm64 Apple M1 Max
  Binaries:
    Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
    npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm
  Utilities:
    Git: 2.39.2 - /usr/bin/git
```

**Comment by plouc** [comment]
Probably related to https://github.com/lerna/lerna/issues/3637.

**Comment by ghiscoding** [comment]
also related to #2402 and #955 

There's currently 2 issues in Lerna's code related to this
1. Lerna only does 1 dependency version replacement, as soon as a package is found it will bump the version on that package but it will stop there (following this order: `dependencies`, then `devDependencies`, then `optionalDependencies` and then `peerDependencies`) and continue with the next package dependency , in other words considering what is shown below, it will only replace the first one with highest priority (in this case `dependencies`) and since it already did the first occurrence of `@nivo/core`  in `dependencies`, it is considered done on that dependency name and will never see/visit the next peer deps one with the same package name
```json
"dependencies": {
    "@nivo/core": "workspace:*"
},
"peerDependencies": {
    "@nivo/core": "workspace:*"
},
```

2. it was identified in Lerna that peer deps should never be updated as per this issue #1018 and later confirmed by Lerna maintainer in the same issue

**Comment by plouc** [comment]
> After all, the workspace prefix should probably never appear in published package.json files.

Yes, that was also my thinking, the workspace protocol doesn't really make sense once the packages are published and not in a workspace anymore.

**Comment by JamesHenry** [comment]
Closing for housekeeping as this shipped a while ago in https://github.com/lerna/lerna/pull/4203
